### PR TITLE
feat(oracles): oracle full profile

### LINKED ISSUE #3227: Show full oracle profile info in markets
## Story

As a trader interested in entering a market
I want evidence of the market oracle's reputation
So I can decide if it is trustworthy

## Acceptance Criteria

- [x] I can open the long oracle profile modal from the short oracle profile
- [x] I can find links to all proofs provided for this oracle
- [x] I can see which other markets (if any) are configured to use this key as an oracle, along with links and details for those markets
- [x] The long profile reflects all oracle statuses

## Tasks

- [x] Implement long oracle profile as a component which can be reused in other apps, like governance
- [x] Pull data from github wellknown repo
- [x] Share a deploy preview link with Neil & Joelle
- [ ] QA

## Sketch

See [the Miro board](https://miro.com/app/board/uXjVOs_PBKk=/?moveToWidget=3458764536859308772&cot=14) for sketches

## Design

designs are in the Governance Figma: [Figma link](https://www.figma.com/file/X9l0t8TZUF1tU8u0ILV3ws/Governance-%26-Token?node-id=2502%3A14216&t=U4WmgcydlTw2i3nf-1)

### Showing oracle status in the info pane

image

### Showing full oracle profile in a modal

See Figma for the other status states for this modal

image

## Additional details / background info

- This ticket is dependent on:
  - the task "Visual design → Long oracle profile" from the epic: https://github.com/vegaprotocol/ux-and-visual-design/issues/79
  - https://github.com/vegaprotocol/frontend-monorepo/issues/2086
- Oracle profiles will be located in the repo here: https://github.com/vegaprotocol/well-known/tree/main/oracle-providers

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #3227 

# Description ℹ️

Oracle full profile component. 

# Demo 📺

Screenshot 2023-05-04 at 11 34 36

Screenshot 2023-04-26 at 01 08 41

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

**Review by pscott31 — APPROVED** [review]

File: libs/oracles/src/lib/hooks/use-oracle-markets.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ?.map((edge) => edge.node)
+    ?.filter((node) => {
+      const p = node.tradableInstrument.instrument.product;
+      const sourceType = p.dataSourceSpecForSettlementData.data.sourceType;
+      if (sourceType.__typename !== 'DataSourceDefinitionExternal') {
+        return false;
+      }
+      const signers = sourceType?.sourceType.signers;
+
+      const signerKeys = signers?.map(
```
Comment (mattrussell36) [inline libs/oracles/src/lib/hooks/use-oracle-markets.ts:24]: suggestion (non blocking): I find this super hard to read. Something like this is much clearer, explicitly filter any undefined values first, its now clearer to see that different properties are used depending on the __typename

```
const signerKeys = signers?.filter(Boolean).map((signer) => {
  if (signer.signer.__typename === 'ETHAddress') {
    return signer.signer.address
  }

  if (signer.signer.__typename === 'PubKey') {
    return signer.signer.key
  }

  return undefined
});


const signedProofsKeys = signedProofs.map((proof) => {
  if ('public_key' in proof && proof.public_key) {
    return true
  }
  if ('eth_address' in proof && proof.eth_address) {
    return true
  }
  return false 
});
```

**Review by mattrussell36 — APPROVED** [review]
Would be good to add some e2e tests

File: libs/market-info/src/components/market-info/market-info-panels.tsx [hunk 2]
```diff
@@ -598,3 +610,36 @@ const NoOracleProof = ({
...
+}) => {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const oracleMarkets = useOracleMarkets(provider);
+  return (
+    <div key={provider.name}>
+      <OracleBasicProfile
+        provider={provider}
+        onClick={() => setDialogOpen(!dialogOpen)}
+        markets={oracleMarkets}
+      ></OracleBasicProfile>
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/market-info/src/components/market-info/market-info-panels.tsx]: updated

File: libs/oracles/src/lib/hooks/use-oracle-markets.ts (hunk 1 above)
Comment (MadalinaRaicu) [reply libs/oracles/src/lib/hooks/use-oracle-markets.ts]: updated

File: libs/oracles/src/lib/components/oracle-full-profile/oracle-full-profile.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    .filter((proof) => proof.format === 'url' && proof.available === true)
+    .map((proof) => ({
+      ...proof,
+      url: 'url' in proof ? proof.url : '',
+      icon: getLinkIcon(proof.type),
+    }));
+
+  return (
+    <div className="flex flex-col text-sm">
+      <p className="dark:text-vega-light-300 text-vega-dark-300 pb-2">
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/oracles/src/lib/components/oracle-full-profile/oracle-full-profile.tsx]: updated

**Pull request merged**